pkg/neonvm/controllers/vm_controller.go

```diff
@@ -1443,6 +1463,21 @@ func podSpec(
 		},
 	}
 
+	if withReadinessProbe {
```

Comment:
this is for the live migration, but I don't like this approach. I pushed that version to discuss what should be the correct approach for the live migrations. The thing is that qemu in target pod doesn't boot until migration is performed. But the migration is not performed unless pod is ready. There are 2 ways to tackle the issue:
- either change ready probe in target pods (the line I left comment for does that) but we can't change readyiness probe on running pod so this pod will be forever without readiness probe. I found this suboptimal tbh - some pods will have probe, some won'y
- change the logic in live migration controller and don't wait for the pod readiness

@sharnoff do you see any better options ?

Reply:
Hm, readiness probes (a) are valid beyond startup, and (b) are used to route service traffic, right?

I would imagine that VM migration doesn't work well with services because we either have extended time with >1 pod ready or a brief moment when the old pod is "ready" but shouldn't receive traffic.

But it'd be a little weird to have readiness probes at runtime only working if the VM hasn't been migrated yet...

Given that, I would probably recommend the second option you gave: use the readiness probe on both pods, and change the logic in the migration controller. That way, we have a bit more information about the migration state visible from the pods, services should work a _little_ better, and we get consistent readiness probes across the lifetime of a VM.